src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc

```diff
@@ -0,0 +1,168 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+
+#include "gtest/gtest.h"
+#include "utils/shared_ptr.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/commands/command.h"
+#include "application_manager/mock_event_dispatcher.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "hmi/vi_read_did_response.h"
+#include "hmi/vi_subscribe_vehicle_data_response.h"
+#include "hmi/vi_get_vehicle_type_response.h"
+#include "hmi/vi_is_ready_response.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+
+using ::testing::_;
+using ::testing::ReturnRef;
+using ::testing::Types;
+using ::testing::Eq;
+
+using ::utils::SharedPtr;
+using ::test::components::event_engine_test::MockEventDispatcher;
+
+namespace am = ::application_manager;
+namespace commands = am::commands;
+using commands::MessageSharedPtr;
+
+template <class CommandD>
+class ResponseFromHMICommandsTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  typedef CommandD CommandData;
+  MockEventDispatcher event_dispatcher_;
+
+  ResponseFromHMICommandsTest() {
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(event_dispatcher_));
+  }
+};
+
+template <class Command, hmi_apis::FunctionID::eType kEventId>
+struct CommandData {
+  typedef Command CommandType;
+  enum { kExpectedEventId = kEventId };
+};
+
+typedef Types<
+    CommandData<commands::VIReadDIDResponse,
+                hmi_apis::FunctionID::VehicleInfo_ReadDID>,
+    CommandData<commands::VISubscribeVehicleDataResponse,
+                hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData> >
+    ResponseCommandsList;
+
+TYPED_TEST_CASE(ResponseFromHMICommandsTest, ResponseCommandsList);
+
+ACTION_P(GetEventFunctionID, function_id) {
+  (*function_id) = arg0.id();
+}
+
+TYPED_TEST(ResponseFromHMICommandsTest, Run_SendMessageToHMI_SUCCESS) {
+  typedef typename TestFixture::CommandData CommandData;
+  typedef typename CommandData::CommandType CommandType;
+
+  hmi_apis::FunctionID::eType kExpectedEventId =
+      static_cast<hmi_apis::FunctionID::eType>(CommandData::kExpectedEventId);
+
+  SharedPtr<CommandType> command = this->template CreateCommand<CommandType>();
+
+  hmi_apis::FunctionID::eType event_id;
+  EXPECT_CALL(this->event_dispatcher_, raise_event(_))
```

Comment:
@vlantonov, yes, it's required, because event_dispatcher_ is inherited from another template class which take one of template parameters of this template class, so it cant know about its fields and methods.